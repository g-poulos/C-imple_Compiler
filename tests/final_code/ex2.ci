program main
{
    declare a,A,B;
    function f1(in a, inout b)
    {
        function f2(in a)
        {
            B:=A;
            b:=a;
        }
        A:=2;
        call f2(in a);

    }
    a:=3;
    A:=4;
    call f1(in a, inout A);
    print(A);
    print(B);
}.