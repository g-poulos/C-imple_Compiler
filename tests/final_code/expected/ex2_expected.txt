L0:
b Lmain
L1:
sw ra, -0(sp)
L2:
lw t1, -16(gp)
sw t1, -20(gp)
L3:
lw t0, -4(sp)
addi t0, t0, -12
lw t1, (t0)
lw t0, -4(sp)
addi t0, t0, -16
lw t0, (t0)
sw t1, (t0)
L4:
lw ra, -0(sp)
jr ra
L5:
sw ra, -0(sp)
L6:
li t1, 2
sw t1, -16(gp)
L7:
addi fp, sp, 12
sw sp, -4(fp)
addi sp, sp, 12
jal L1
addi sp, sp, -12
L8:
lw ra, -0(sp)
jr ra
Lmain:
L9:
addi sp, sp, 24
move gp, sp
L10:
li t1, 3
sw t1, -12(sp)
L11:
li t1, 4
sw t1, -16(sp)
L12:
addi fp, sp, 20
lw t1, -12(sp)
sw t1, -12(fp)
L13:
addi t0, sp, -16
sw t0, -16(fp)
L14:
sw sp, -4(fp)
addi sp, sp, 20
jal L5
addi sp, sp, -20
